hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/DataInputNavigableJsonAdapterTest.java

```diff
@@ -110,10 +110,23 @@ public void testMalformedSamples()
         }
     }
 
-    // ensure that when UTF8Reader instances are created & used from several threads safely
-    // UTF8Reader is not thread safe, but caches CharsetDecoders in ThreadLocals
     @Test
-    public void testConcurrently() throws InterruptedException {
+    public void testMixedRead_whenMultiReadFirst() throws IOException {
+        for (byte[] sample : SAMPLE_BYTES) {
+            assertReadMixed(sample, false);
+        }
+    }
+
+    @Test
+    public void testMixedRead_whenSingleReadFirst() throws IOException {
+        for (byte[] sample : SAMPLE_BYTES) {
+            assertReadMixed(sample, true);
+        }
+    }
+
+    @Test
+    public void testCreateAndUseConcurrently() throws InterruptedException {
+        // UTF8Reader is not thread safe; the purpose is to ensure no
```

Comment:
minor: "the purpose is to ensure no" - no what?

Reply:
🤦‍♂ too hasty to push